PyPortal_Alarm_Clock/code.py

```diff
@@ -0,0 +1,506 @@
+
+        # each button has it's edges as well as the state to transition to when touched
+        self.buttons = [dict(left=0, top=50, right=80, bottom=120, next_state='settings'),
+                        dict(left=0, top=155, right=80, bottom=220, next_state='mugsy')]
+
+
+    @property
+    def name(self):
+        return 'time'
+
+
+    def adjust_backlight_based_on_light(self, force=False):
+        """Check light level. Adjust the backlight and background image if it's dark."""
+        global low_light
+        if light.value <= 1000 and (force or not low_light):
+            pyportal.set_backlight(0.01)
+            pyportal.set_background(self.background_night)
+            low_light = True
+        elif force or (light.value >= 2000 and low_light):
+            pyportal.set_backlight(1.00)
+            pyportal.set_background(self.background_day)
+            low_light = False
+
+
+    def tick(self, now):
+        global alarm_armed, snooze_time, update_time
+
+        # is the snooze button pushed? Cancel the snooze if so.
+        if not snooze_button.value:
+            if snooze_time:
+                self.snooze_icon.pop()
+            snooze_time = None
+            alarm_armed = False
+
+        # is snooze active and the snooze time has passed? Transition to alram is so.
+        if snooze_time and ((now - snooze_time) >= snooze_interval):
+            change_to_state('alarm')
+            return
+
+        # check light level and adjust background & backlight
+        #self.adjust_backlight_based_on_light()
+
+        # only query the online time once per hour (and on first run)
+        if (not self.refresh_time) or ((now - self.refresh_time) > 3600):
+            try:
+                pyportal.get_local_time(location=secrets['timezone'])
+                self.refresh_time = now
+            except RuntimeError as e:
+                logger.error('Some error occured, retrying! - %s', str(e))
+
+        # only query the weather every 10 minutes (and on first run)
+        if (not self.weather_refresh) or (now - self.weather_refresh) > 600:
+            try:
+                value = pyportal.fetch()
+                weather = json.loads(value)
+
+                # set the icon/background
+                weather_icon_name = weather['weather'][0]['icon']
+                try:
+                    self.weather_icon.pop()
+                except IndexError:
+                    pass
+                filename = "/icons/"+weather_icon_name+".bmp"
+                if filename:
+                    if self.icon_file:
+                        self.icon_file.close()
+                    self.icon_file = open(filename, "rb")
+                    icon = displayio.OnDiskBitmap(self.icon_file)
+                    icon_sprite = displayio.TileGrid(icon,
+                                                     pixel_shader=displayio.ColorConverter(),
+                                                     position=(0, 0))
```

Comment:
This will no loner work with latest version of displayio - instead of position it uses x, y